### PR TITLE
Fix `collinear` and `segment::contains`

### PR DESCRIPTION
**Pull request by jsn0716**
### Description

Fix `collinear` and `segment::contains` that is breaking tests for #1230. Removed the old collinear as well.

### Testing Done

Passes tests when merged with #1230, and added unit tests that covers the failure.

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by scveloso — DISMISSED** [review]
LGTM thanks for the quick fix on this!

File: src/software/new_geom/segment.h [hunk 1]
```diff
@@ -89,9 +90,10 @@ class Segment final
         {
             // If the segment and point are in a perfect vertical line, we must use Y
             // coordinate centric logic
-            if ((std::abs(point.x() - getEnd().x()) < GeomConstants::FIXED_EPSILON) &&
-                (std::abs(getEnd().x() - getSegStart().x()) <
-                 GeomConstants::FIXED_EPSILON))
+            if (almostEqual(point.x(), getEnd().x(), GeomConstants::FIXED_EPSILON,
+                            GeomConstants::ULPS_EPSILON_TEN) &&
+                almostEqual(getEnd().x(), getSegStart().x(), GeomConstants::FIXED_EPSILON,
+                            GeomConstants::ULPS_EPSILON_TEN))
```
Comment (garethellis0) [inline src/software/new_geom/segment.h]: Hmm, the epsilon to use should be decided on by the situation it's used in. Should we expose this as an argument for `colinear`? Then it really should be propogated up to a point where we have some intuition for what reasonable epsilon values would be?

Comment (scveloso) [reply src/software/new_geom/segment.h]: Ah I see what you mean, maybe introduce a default parameter EPSILON? so for situations where we need to loosen it, we add an extra EPSILON argument?

Comment (garethellis0) [reply src/software/new_geom/segment.h]: Yes, IMO that parameter should be present on pretty much all geom functions, but in the interests of keeping this in the scope of this PR, maybe let's just add it to any functions we modified?

Comment (jsn0716) [reply src/software/new_geom/segment.h]: Added extra arguments with defaults to geom util functions

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by scveloso — APPROVED** [review]
LGTM!

**Pull request merged**